impeller/entity/contents/filters/gaussian_blur_filter_contents.cc

```diff
@@ -255,6 +236,50 @@ Entity ApplyBlurStyle(FilterContents::BlurStyle blur_style,
       }));
   return result;
 }
+
+Entity ApplyBlurStyle(FilterContents::BlurStyle blur_style,
+                      const Entity& entity,
+                      const std::shared_ptr<FilterInput>& input,
+                      const Snapshot& input_snapshot,
+                      Entity blur_entity,
+                      const std::shared_ptr<Geometry>& geometry) {
+  switch (blur_style) {
+    case FilterContents::BlurStyle::kNormal:
+      return blur_entity;
+    case FilterContents::BlurStyle::kInner:
+      return ApplyClippedBlurStyle(Entity::ClipOperation::kIntersect, entity,
+                                   input, input_snapshot,
+                                   std::move(blur_entity), geometry);
+      break;
+    case FilterContents::BlurStyle::kOuter:
+      return ApplyClippedBlurStyle(Entity::ClipOperation::kDifference, entity,
+                                   input, input_snapshot,
+                                   std::move(blur_entity), geometry);
+    case FilterContents::BlurStyle::kSolid: {
+      Entity blurred = ApplyClippedBlurStyle(Entity::ClipOperation::kIntersect,
+                                             entity, input, input_snapshot,
+                                             std::move(blur_entity), geometry);
+      Entity snapshot_entity = Entity::FromSnapshot(
+          input_snapshot, entity.GetBlendMode(), entity.GetClipDepth());
+      Entity result;
+      result.SetContents(Contents::MakeAnonymous(
+          fml::MakeCopyable([blurred = blurred.Clone(),
+                             snapshot_entity = std::move(snapshot_entity)](
+                                const ContentContext& renderer,
+                                const Entity& entity,
+                                RenderPass& pass) mutable {
+            bool did_render = true;
+            did_render = blurred.Render(renderer, pass) && did_render;
+            did_render = snapshot_entity.Render(renderer, pass) && did_render;
+            return did_render;
```

Comment:
```suggestion
            return blurred.Render(renderer, pass) && snapshot_entity.Render(renderer, pass);
```

Reply:
That's not functionally equivalent because of short-circuiting.  I was thinking we still want to render the snapshot even if the blurred draw fails for some reason.

Reply:
If part of rendering fails then we should almost always be bailing out to a toplevel VALIDATION_LOG handler or error. partially correct is still wrong, so I'd prefer the simpler approach.

Reply:
I think those validation errors are noops in release builds though, right?  Not a huge deal, done.